PoGo.NecroBot.Logic/Tasks/SnipePokemonTask.cs

```diff
@@ -79,6 +79,14 @@ public static class SnipePokemonTask
                     {
                         foreach (var pokemonLocation in locationsToSnipe)
                         {
+                            var pokeBallsCount = await session.Inventory.GetItemAmountByType(POGOProtos.Inventory.Item.ItemId.ItemPokeBall);
+                            var greatBallsCount = await session.Inventory.GetItemAmountByType(POGOProtos.Inventory.Item.ItemId.ItemGreatBall);
+                            var ultraBallsCount = await session.Inventory.GetItemAmountByType(POGOProtos.Inventory.Item.ItemId.ItemUltraBall);
+                            var masterBallsCount = await session.Inventory.GetItemAmountByType(POGOProtos.Inventory.Item.ItemId.ItemMasterBall);
+
+                            if (pokeBallsCount + greatBallsCount + ultraBallsCount + masterBallsCount < session.LogicSettings.MinPokeballsToSnipe)
+                                return;
+
```

Comment:
The only way to solve that is to randomize which locations to do or to make sure your pokeball count is high enough to make it through all. TBH after a few iterations the first stops will clear out somewhat and you'll make it to the next.